src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -2886,7 +2886,9 @@ private sealed class MemberDescriptorComparer : IComparer
 
             public int Compare(object left, object right)
             {
-                return CultureInfo.InvariantCulture.CompareInfo.Compare(((MemberDescriptor)left).Name, ((MemberDescriptor)right).Name);
+                MemberDescriptor leftMember = left as MemberDescriptor;
+                MemberDescriptor rightMember = right as MemberDescriptor;
+                return CultureInfo.InvariantCulture.CompareInfo.Compare(leftMember?.Name, rightMember?.Name);
```

Comment:
Not sure if this is the right way to do this - should we throw if we encounter a `non-MemberDescriptor` object in the list?

Reply:
It would be an improvement to throw when object is not MemberDescriptor.
We could add that in a follow up PR too.